geode-redis/src/main/java/org/apache/geode/redis/internal/executor/CommandFunction.java

```diff
@@ -50,152 +47,115 @@ public static void register() {
     FunctionService.registerFunction(new CommandFunction(stripedExecutor));
   }
 
+
   @SuppressWarnings("unchecked")
-  public static ResultCollector execute(Region<?, ?> region,
-      RedisCommandType command,
+  public static <T> T execute(RedisCommandType command,
       ByteArrayWrapper key,
-      Object commandArguments) {
-    return FunctionService
+      Object commandArguments, Region region) {
+    SingleResultCollector<T> rc = new SingleResultCollector<>();
+    ResultCollector<T, T> execute = FunctionService
         .onRegion(region)
         .withFilter(Collections.singleton(key))
         .setArguments(new Object[] {command, commandArguments})
-        .execute(ID);
+        .withCollector(rc)
+        .execute(CommandFunction.ID);
+    return execute.getResult();
   }
 
 
   public CommandFunction(StripedExecutor stripedExecutor) {
     this.stripedExecutor = stripedExecutor;
   }
 
-  @SuppressWarnings("unchecked")
   @Override
-  public void execute(FunctionContext<Object[]> context) {
-    RegionFunctionContextImpl regionFunctionContext =
-        (RegionFunctionContextImpl) context;
-    ByteArrayWrapper key =
-        (ByteArrayWrapper) regionFunctionContext.getFilter().iterator().next();
-    Region localRegion =
-        regionFunctionContext.getLocalDataSet(regionFunctionContext.getDataSet());
-    ResultSender resultSender = regionFunctionContext.getResultSender();
-    Object[] args = context.getArguments();
-    RedisCommandType command = (RedisCommandType) args[0];
+  public String getId() {
+    return ID;
+  }
+
+  @Override
+  protected Object compute(Region localRegion, ByteArrayWrapper key,
+      RedisCommandType command, Object[] args) {
+    Callable<Object> callable;
     switch (command) {
       case SADD: {
         ArrayList<ByteArrayWrapper> membersToAdd = (ArrayList<ByteArrayWrapper>) args[1];
-        stripedExecutor.execute(key,
-            () -> RedisSet.sadd(localRegion, key, membersToAdd),
-            (addedCount) -> resultSender.lastResult(addedCount));
+        callable = () -> new Long(RedisSet.sadd(localRegion, key, membersToAdd));
```

Comment:
I think Long.valueOf is preferred over new Long. It has the benefit of returning a cached instance of Longs in the range of -128..127 which is probably what we have most of the time in this code.

Reply:
yea, this is just a relic of me trying to debug stuff and trying dumb ideas...i removed the boxing.